internal/evaluator/eval.go

```diff
@@ -129,13 +129,18 @@ func (e *Evaluator) evaluateCondition(ctx *hcl.EvalContext, content *hcl.BodyCon
 		if val.Type() != cty.Bool {
 			return false, diags.Append(hclutils.Err2Diag(fmt.Errorf("got type %s, expected %s", val.Type(), cty.Bool)))
 		}
-		if !val.True() {
+		shouldDiscard := !val.IsKnown()
+		if !shouldDiscard {
+			shouldDiscard = !val.True()
+		}
+		if shouldDiscard {
 			e.discard(DiscardItem{
 				Type:        et,
 				Reason:      discardReasonUserCondition,
```

Comment:
the reason needs to be different for incomplete values because we drop user reason discards from output. So if the value is explicitly false, it should be discarded as a user condition and if it is incomplete, it should be an implicit discard which will appear in events, conditions etc.

Reply:
that makes sense, addressed in https://github.com/crossplane-contrib/function-hcl/pull/26/commits/f734a251cdd40c5a5422db584d24dcf824b7bb61

Reply:
Updated the logic in https://github.com/crossplane-contrib/function-hcl/pull/26/commits/66869fe53c898d2595f5d772c6fdcf0ef078b8b0 to be like what we do in other places. I also moved a test as semantic changed. This is hopefully correct now.